src/WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -871,14 +872,23 @@ private static ComInterfaceEntry ProvideIReferenceArray(Type arrayType)
 
         internal sealed class InspectableInfo
         {
-            private readonly Lazy<string> runtimeClassName;
+            private readonly Type _type;
 
             public Guid[] IIDs { get; }
-            public string RuntimeClassName => runtimeClassName.Value;
+
+            private volatile string _runtimeClassName;
+
+            private string MakeRuntimeClassName()
+            {
+                global::System.Threading.Interlocked.CompareExchange(ref _runtimeClassName, TypeNameSupport.GetNameForType(_type, TypeNameGenerationFlags.GenerateBoxedName | TypeNameGenerationFlags.ForGetRuntimeClassName), null);
+                return _runtimeClassName;
+            }
+
+            public string RuntimeClassName => _runtimeClassName ?? MakeRuntimeClassName();

```

Comment:
Nit: can simplify this by just moving the code for `MakeRuntimeClassName` in the getter (using a body), rather than needing the separate method there, which is only ever used here anyway?

Reply:
This is the pattern followed elsewhere so sticking with that.

Reply:
But it's... Not a great pattern? And it's also wasteful (more IL size and extra JIT work for no reason), not to mention unnecessarily verbose. We should try to improve the codebase as we find things, rather than just sticking to existing code patterns that are not good in the first place. Of course, this can be a follow up 🙂

Reply:
I'm not saying that we shouldn't change it but I'd rather change it everywhere at the same time :).